portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro

```diff
@@ -10,7 +10,14 @@ definition {
 
 		Variables.assertDefined(parameterList = "${typeSettings}");
 
-		var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+		if (isSet(remoteAppEntryId)) {
+			var widget = JSONLayoutUtil._generateWidget(
+				remoteAppEntryId = "${remoteAppEntryId}",
+				widgetName = "${widgetName}");
+		}
```

Comment:
I don't quite like adding remoteAppEntryID into _generateWidget. It's expecting 		Variables.assertDefined(parameterList = "${widgetName}"); But doesn't need remoteAppEntryID.

Maybe instead, right after _generateWidget, we can check if (isSet(remoteAppEntryId)) and append the remoteAppEntryID.